server/memphis_handler_schemas.go

```diff
@@ -0,0 +1,368 @@
+			err = schemasVersionCollection.FindOne(context.TODO(), filter).Decode(&schemaVersion)
+			if err != nil {
+				return []models.ExtendedSchema{}, err
+			}
+			if schemaVersion.VersionNumber == 1 {
+				extSchema := models.ExtendedSchema{
+					ID:            schema.ID,
+					Name:          schema.Name,
+					Type:          schema.Type,
+					CreatedByUser: schemaVersion.CreatedByUser,
+					CreationDate:  schemaVersion.CreationDate,
+				}
+				extednedSchemas = append(extednedSchemas, extSchema)
+			}
+		}
+
+	}
+
+	if len(extednedSchemas) == 0 {
+		return []models.ExtendedSchema{}, nil
+	} else {
+		return extednedSchemas, nil
+	}
+}
+
+func (sh SchemasHandler) GetAllSchemaVersionsByVersionNumber(versionNumber int) ([]models.SchemaVersion, error) {
+	var schemasVersion []models.SchemaVersion
+
+	filter := bson.M{"version_number": versionNumber}
+	cursor, err := schemasVersionCollection.Find(context.TODO(), filter)
+	if err != nil {
+		return []models.SchemaVersion{}, err
+	}
+	if err = cursor.All(context.TODO(), &schemasVersion); err != nil {
+		return schemasVersion, err
+	}
+
+	if err != nil {
+		return []models.SchemaVersion{}, err
+	}
+
+	if len(schemasVersion) == 0 {
+		return []models.SchemaVersion{}, nil
+	} else {
+		return schemasVersion, nil
+	}
+}
+
+func (sh SchemasHandler) FindAndDeleteSchema(schemaName string) error {
+	var schema models.Schema
+	filter := bson.M{"name": schemaName}
+	err := schemasCollection.FindOne(context.TODO(), filter).Decode(&schema)
+	if err != nil {
+		return err
+	}
+	schemaVersions := schema.Versions
+	var schemaVersion models.SchemaVersion
+	for _, version := range schemaVersions {
+		filterVersion := bson.M{"_id": version}
+		err = schemasVersionCollection.FindOneAndDelete(context.TODO(), filterVersion).Decode(&schemaVersion)
+		if err != nil {
+			return err
+		}
+	}
+	_, err = schemasCollection.DeleteOne(context.TODO(), filter)
+
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sh SchemasHandler) CreateNewSchema(c *gin.Context) {
+	var body models.CreateNewSchema
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	schemaName := strings.ToLower(body.Name)
+	err := validateSchemaName(schemaName)
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		return
+	}
+	exist, _, err := IsSchemaExist(schemaName)
+	if exist {
+		serv.Warnf("Schema with that name already exists")
+	}
+	if err != nil {
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("CreateNewSchema error: " + err.Error())
+		c.AbortWithStatusJSON(401, gin.H{"message": "Unauthorized"})
+	}
+	schemaType := strings.ToLower(body.Type)
+	err = validateSchemaType(schemaType)
+
+	if err != nil {
+		serv.Warnf(err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": err.Error()})
+		return
+	}
+	schemaContent := strings.ToLower(body.SchemaContent)
```

Comment:
you don't need to switch to lower the user schema this is the schema he defined for his data, leave it as is